skrifa/src/outline/mod.rs

```diff
@@ -0,0 +1,448 @@
+            OutlineCollectionKind::Glyf(glyf)
+        } else if let Ok(cff) = cff::Outlines::new(font) {
+            OutlineCollectionKind::Cff(cff)
+        } else {
+            OutlineCollectionKind::None
+        };
+        Self { kind }
+    }
+
+    /// Creates a new outline collection for the given font and outline
+    /// format.
+    ///
+    /// Returns `None` if the font does not contain outlines in the requested
+    /// format.
+    pub fn with_format(font: &impl TableProvider<'a>, format: OutlineGlyphFormat) -> Option<Self> {
+        let kind = match format {
+            OutlineGlyphFormat::Glyf => OutlineCollectionKind::Glyf(glyf::Outlines::new(font)?),
+            OutlineGlyphFormat::Cff => {
+                let upem = font.head().ok()?.units_per_em();
+                OutlineCollectionKind::Cff(cff::Outlines::from_cff(font.cff().ok()?, 0, upem).ok()?)
+            }
+            OutlineGlyphFormat::Cff2 => {
+                let upem = font.head().ok()?.units_per_em();
+                OutlineCollectionKind::Cff(cff::Outlines::from_cff2(font.cff2().ok()?, upem).ok()?)
+            }
+        };
+        Some(Self { kind })
+    }
+
+    /// Returns the underlying format of the source outline tables.
+    pub fn format(&self) -> Option<OutlineGlyphFormat> {
+        match &self.kind {
+            OutlineCollectionKind::Glyf(..) => Some(OutlineGlyphFormat::Glyf),
+            OutlineCollectionKind::Cff(cff) => cff
+                .is_cff2()
+                .then_some(OutlineGlyphFormat::Cff2)
+                .or(Some(OutlineGlyphFormat::Cff)),
+            _ => None,
+        }
+    }
+
+    /// Returns the outline for the given glyph identifier.
+    pub fn get(&self, glyph_id: GlyphId) -> Option<OutlineGlyph<'a>> {
+        match &self.kind {
+            OutlineCollectionKind::None => None,
+            OutlineCollectionKind::Glyf(glyf) => Some(OutlineGlyph {
+                kind: OutlineKind::Glyf(glyf.clone(), glyf.outline(glyph_id).ok()?),
+            }),
+            OutlineCollectionKind::Cff(cff) => Some(OutlineGlyph {
+                kind: OutlineKind::Cff(cff.clone(), glyph_id, cff.subfont_index(glyph_id)),
+            }),
+        }
+    }
+
+    /// Returns an iterator over all of the outline glyphs in the collection.
+    pub fn iter(&self) -> impl Iterator<Item = (GlyphId, OutlineGlyph<'a>)> + 'a + Clone {
+        let len = match &self.kind {
+            OutlineCollectionKind::Glyf(glyf) => glyf.len(),
+            OutlineCollectionKind::Cff(cff) => cff.len(),
+            _ => 0,
+        } as u16;
+        let copy = self.clone();
+        (0..len).filter_map(move |gid| {
+            let gid = GlyphId::new(gid);
+            let glyph = copy.get(gid)?;
+            Some((gid, glyph))
+        })
+    }
+}
+
+#[derive(Clone)]
+enum OutlineCollectionKind<'a> {
+    None,
+    Glyf(glyf::Outlines<'a>),
+    Cff(cff::Outlines<'a>),
+}
+
+const GLYF_SCALER_STACK_BUFFER_SIZE: usize = 4096;
```

Comment:
Comment should explain why this value

Reply:
Noted. I have actual data now for memory requirements with TT hinting so this is likely to become more complicated with multiple stack allocation sizes. I’ll prep a sheet with this data because it’s rather interesting.

Reply:
Admitted _arbitrariness_ of this constant. Will make this more clear in the hinting PR.